src/main/java/games/strategy/engine/lobby/server/userDB/DbUserController.java

```diff
@@ -0,0 +1,112 @@
+package games.strategy.engine.lobby.server.userDB;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Preconditions;
+
+
+/**
+ * Interface for interacting with database user operations. For example, add user, delete user..
+ */
+/*
+ * TODO: datasource migration note, from derby (java DB) to MySQL
+ *  This class is currently set up to write and read from multiple data sources.
+ *  This is a migration tool, to go from one data source to another. When we switch over completely
+ *  we can simplify and write/read to the primary datasource directly.
+ */
+public class DbUserController implements UserDao {
+  private static final Logger logger = Logger.getLogger(DbUserController.class.getName());
+
+  private static final Collection<UserDaoPrimarySecondary> implementations = Arrays.asList(
+      new DerbyUserController(),
+      new SqlUserController()
+  );
+
+  /**
+   * Stores a convenience reference to the authoritative DAO implementation.
+   * We use this datasource for reading, we always write to it, any errors during read/write are considered
+   * critical (on the other hand secondary DAO read/write errors can be logged and ignored)
+   */
+  private static final UserDao primaryDao;
+
+
+  static {
+    primaryDao = determinePrimaryDataSource();
+  }
+
+
+  private static UserDao determinePrimaryDataSource() {
+    List<UserDao> primaryDataSourceCandidates = implementations.stream()
+        .filter(UserDaoPrimarySecondary::isPrimary)
+        .collect(Collectors.toList());
+
+    Preconditions.checkState(
+        primaryDataSourceCandidates.size() == 1,
+        "Startup Assumption - exactly one UserDao implementation is 'primary'");
+
+    return primaryDataSourceCandidates.get(0);
+  }
+
+
+  /**
+   * @return null if the user does not exist.
+   */
+  @Override
+  public String getPassword(final String userName) {
+    return primaryDao.getPassword(userName);
+  }
+
+  @Override
+  public boolean doesUserExist(final String userName) {
+    return primaryDao.doesUserExist(userName);
+  }
+
+  @Override
+  public void updateUser(final String name, final String email, final String hashedPassword, final boolean admin) {
+    primaryDao.updateUser(name, email, hashedPassword, admin);
+    try {
+      implementations.forEach(dao -> dao.updateUser(name, email, hashedPassword, admin));
+    } catch(Exception e) {
```

Comment:
I think the answer is yes, it is appropriate. Let me clarify the requirements I was trying to satisfy.

1. Any calls to `updateUser` would have the same effect as if `primaryDao.updateUser` were called directly. In this scenario `primaryDao` would throw an IllegalStateException and that would be uncaught and would bubble upt to the caller. The important piece of code to keep working is here: https://github.com/triplea-game/triplea/blob/master/src/main/java/games/strategy/engine/lobby/server/UserManager.java#L31  . Side note, we are using IllegalState to mostly flag when a user does not exist, and this also mixes the concept of DB failure overall. The interaction with UserManager would be a good thing to fix up, but hence why the IllegalState is so wonky.

2. Second requirement, which this try/catch fulfills, any failures on the secondary are to be ignored. We will need to be careful about timeouts, and may want to make these calls on a background thread. I'll add a TODO. comment for that.

Please let me know if that answers your questions, happy to follow up further/clarify.